app/template/rest.py

```diff
@@ -249,10 +249,18 @@ def preview_letter_template_by_notification_id(service_id, notification_id, file
                     status_code=500
                 )
 
-            url = '{}/precompiled-preview.png{}'.format(
-                current_app.config['TEMPLATE_PREVIEW_API_HOST'],
-                '?hide_notify=true' if page_number == '1' else ''
-            )
+            if request.args.get('overlay'):
+                content = pdf_page
+                url = '{}/precompiled/overlay.png{}{}'.format(
+                    current_app.config['TEMPLATE_PREVIEW_API_HOST'],
+                    '?invert=1',
+                    '&hide_notify=true' if page_number == '1' else ''
```

Comment:
it doesn't look like overlay.png uses `hide_notify`, so that can be removed (and the invert bit inlined in the orig format string).

if we do intend to make that change at some point soon, then we could use `urllib.parse.urlencode` to create query strings from a dictionary and clean this function up a bit.

We could even go the whole hog and properly split it up and construct a `ParseResult` object

Reply:
I got rid of tag hiding arg, the rest sounds interesting although cryptic to me, I'd love to explore it together on Wednesday if you are up for it :)